autoria/api.py

```diff
@@ -334,34 +334,61 @@ def __init__(self, category: str, mark: str, model: str,
         # Getting list of categories and selecting needed id
         category_id = select_item(category, self._api.get_categories())
         mark_id = select_item(mark, self._api.get_marks(category_id))
-        state_id = select_item(state, self._api.get_states())
+        model_id = select_item(
+            model,
+            self._api.get_models(category_id, mark_id)
+        )
 
-        self._params = {
-            'main_category': category_id,
-            'body_id': select_item(
+        state_id, body_id, city_id, gear_id, option_ids, \
+            fuel_ids, drive_ids, color_id = (None,) * 8  # 8 - qty of variables
```

Comment:
This is fun, but I wouldn't use it in real code, probably I would do it like this:

```
self._params = {
           ...
           'state_id': None,
           'body_id': None,
           ...
}

if state is not None:
    self._params['state_id'] = select_item(state, self._api.get_states()

```

This was we don't need additional variables for state_id, body_id, etc (and we don't need to initialize them with Nones).